admin-dev/themes/new-theme/js/pages/product/components/categories/types/index.d.ts

```diff
@@ -0,0 +1,17 @@
+type Category = {
+  id: number,
+  name: string,
+}
+
+type TypeaheadCategory = {
+  id: number,
+  name: string,
+  breadcrumb: string,
+}
+
+type TreeCategory = {
+  id: number,
+  name: string,
+  active: boolean,
+  children: Array<TreeCategory>,
+}
```

Comment:
This is a managing question, do you think that every components folders should have a types folder or should we centralize it inside the root types folder?

Reply:
It depends if type is reused or specifically dedicated for a component. For simplicity I've put these under the same hood with component that uses them.